ocaml/xapi/xapi_pci_helpers.ml

```diff
@@ -172,3 +173,68 @@ let get_host_pcis () =
 let igd_is_whitelisted ~__context pci =
   let vendor_id = Db.PCI.get_vendor_id ~__context ~self:pci in
   List.mem vendor_id !Xapi_globs.igd_passthru_vendor_whitelist
+
+let is_pci_hidden_cmdline ~__context ~self =
+  let cmdline =
+    match Unixext.read_lines ~path:"/proc/cmdline" with
+    | [x] ->
+        x
+    | _ ->
+        failwith "Unable to read cmdline"
+  in
+  let device = Db.PCI.get_pci_id ~__context ~self in
+  let elems = String.split_on_char ' ' cmdline in
+
+  let pciback =
+    List.find_opt
+      (fun s -> String.starts_with ~prefix:"xen-pciback.hide=" s)
+      elems
+  in
+
+  let contains s1 s2 =
+    let re = Re.Str.regexp_string s2 in
```

Comment:
The it would good to document what is the substring and what is the string where the substring is being searched, Is this different from this?
```
let contains str substr = Astring.String.is_infix ~affix:substr str
```

Reply:
I'll add a comment and use your suggestion!